processor/src/main/kotlin/dev/zacsweers/autoservice/ksp/AutoServiceSymbolProcessor.kt

```diff
@@ -82,8 +83,13 @@ public class AutoServiceSymbolProcessor(environment: SymbolProcessorEnvironment)
               return emptyList()
             }
 
-    resolver
-        .getSymbolsWithAnnotation(AUTO_SERVICE_NAME)
+    val symbols = resolver.getSymbolsWithAnnotation(AUTO_SERVICE_NAME)
+
+    val result = symbols.filter { !it.validate() }
```

Comment:
It looks like `validate()` is a fairly greedy check and checks all referenced symbols of a declaration by default. Since we're only interested in supertypes and the autoservice annotation value, can we pass in a lambda that only checks those nodes? Or just write our own custom `KSValidateVisitor` subclass that visits just those.